app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java

```diff
@@ -97,112 +110,115 @@ public void onClick(View view) {
                 if (habitValidator.isHabitValid(habitTitle,
                                                 habitReason,
                                                 habitDate,
-                                                tracker)){
+                                                _tracker)){
                     _habitListFragment.addHabitToDatabase(habitTitle,
                                                           habitReason,
                                                           habitValidator.checkHabitDateValid(habitDate),
-                                                          tracker);
+                                                          _tracker);
                     getDialog().dismiss();
                 }
             }
         });
 
+
+        return view;
+    }
+
+    private void setListenersForDaysOfTheWeek(){
```

Comment:
This is the new setter for the buttons. sets all of the listeners and formats the buttons as necessary.